modules/react/common/spec/components.test-d.tsx

```diff
@@ -0,0 +1,51 @@
+import React from 'react';
+import {expectTypeOf} from 'expect-type';
```

Comment:
[expect-type](https://github.com/mmkal/ts/tree/master/packages/expect-type) is a simple no-dependency package that has no runtime and instead uses Typescript to static analyze with an expect-style matching. It isn't perfect, but it works without a complicated setup. Other contenders were [dtslint](https://github.com/microsoft/dtslint) and [tsd](https://github.com/SamVerschueren/tsd) both which were complicated to set up and had a large amount of dependencies.

It would be nice if `expect-type` failed more gracefully. It can be used as more of a regression-style test. Failures result in some complicated type errors or don't fail correctly. It worked well for enough cases to be useful. It is a very small dependency (basically a 135 TS file) and uses `tsc` to work.